panic-ing on generate if some dependencies have non-gomod module.

Masterminds/squirrel has go.mod, but lann/builder isn't.
goen fails to parse deps like below.

```
/usr/src/goen/example # cat entity.go
package example

import (
        "time"

        "github.com/Masterminds/squirrel"
        uuid "github.com/satori/go.uuid"
)

var _ squirrel.Sqlizer = squirrel.Eq{}

//go:generate goen -o goen.go
type Blog struct {
        BlogID uuid.UUID `goen:"" table:"blogs" primary_key:""`

        Name string

        Author string

        Posts []*Post `foreign_key:"blog_id"`
}

type Post struct {
        Timestamp

        BlogID uuid.UUID `goen:"" table:"posts"`

        PostID int `primary_key:",omitempty"`

        Title string

        Content string

        // "order" is a sql keyword
        Order int

        Blog *Blog `foreign_key:"blog_id"`
}

type Timestamp struct {
        CreatedAt time.Time
        UpdatedAt time.Time
        DeletedAt *time.Time
}
```

```
/usr/src/goen/example # go generate .
2019/11/13 06:40:58 generator.go:143: analyzing package "example"
2019/11/13 06:40:58 generator.go:214: analyzing file "postgres_test.go"
2019/11/13 06:40:58 generator.go:214: analyzing file "sqlite3_test.go"
2019/11/13 06:40:58 generator.go:214: analyzing file "entity.go"
2019/11/13 06:40:58 generator.go:240: analyzing struct type "Blog"
2019/11/13 06:40:58 generator.go:281: analyzing struct field Blog.BlogID as column
panic: goen: unable to assume package name "github.com/Masterminds/squirrel": type-checking package "github.com/Masterminds/squirrel" failed (/go/pkg/mod/github.com/!masterminds/squirrel@v1.1.0/case.go:7:2: could not import github.com/lann/builder (type-checking package "github.com/lann/builder" failed (/go/pkg/mod/github.com/lann/builder@v0.0.0-20180802200727-47ae307949d0/builder.go:5:2: could not import github.com/lann/ps (cannot find package "github.com/lann/ps" in any of:
        /usr/local/go/src/github.com/lann/ps (from $GOROOT)
        /go/src/github.com/lann/ps (from $GOPATH)))))

goroutine 1 [running]:
github.com/kamichidu/goen/internal/asts.AssumePkgName(0xc0003d95c1, 0x1f, 0xa4b12a, 0x1)
        /usr/src/goen/internal/asts/asts.go:107 +0x153
github.com/kamichidu/goen/internal/asts.FindPkgPath(0xc000178300, 0xc0003c7f1c, 0x4, 0x9b6cb, 0x9b6cc)
        /usr/src/goen/internal/asts/asts.go:92 +0x164
github.com/kamichidu/goen/internal.(*aType).Kind.func1(0xb31b60, 0xc0003e3d40, 0x203000, 0x203000)
        /usr/src/goen/internal/structs_a.go:199 +0x2b7
github.com/kamichidu/goen/internal/asts.VisitorFunc.Visit(0xc000372100, 0xb31b60, 0xc0003e3d40, 0x8, 0x40d109)
        /usr/src/goen/internal/asts/asts.go:14 +0x3a
go/ast.Walk(0xb2dc80, 0xc000372100, 0xb31b60, 0xc0003e3d40)
        /usr/local/go/src/go/ast/walk.go:52 +0x66
github.com/kamichidu/goen/internal.(*aType).Kind(0xc0003720e0, 0xc00026b5b0)
        /usr/src/goen/internal/structs_a.go:220 +0xca
github.com/kamichidu/goen/generator.(*Generator).safePkgImport(0xc00026bef8, 0xb3de00, 0xc0003720e0, 0x7, 0x2, 0x3, 0x4)
        /usr/src/goen/generator/generator.go:250 +0x74
github.com/kamichidu/goen/generator.(*Generator).walkStruct(0xc000c4def8, 0xb376c0, 0xc000381be0, 0x1, 0x1)
        /usr/src/goen/generator/generator.go:287 +0x410
github.com/kamichidu/goen/generator.(*Generator).walkFile(0xc000c4def8, 0xc0004bb410, 0xc000178300, 0x1, 0x1)
        /usr/src/goen/generator/generator.go:241 +0x258
github.com/kamichidu/goen/generator.(*Generator).walkPkg(0xc000c4def8, 0xc0004bb410, 0xc00026bac8, 0x1)
        /usr/src/goen/generator/generator.go:216 +0x1e0
github.com/kamichidu/goen/generator.(*Generator).ParseDir(0xc000c4def8, 0x1, 0x1)
        /usr/src/goen/generator/generator.go:151 +0x1db
main.run(0xb2d6a0, 0xc000098000, 0xb2d6e0, 0xc000098008, 0xb2d6e0, 0xc000098010, 0xc000084150, 0x3, 0x3, 0x0)
        /usr/src/goen/cmd/goen/main.go:172 +0x73f
main.main()
        /usr/src/goen/cmd/goen/main.go:208 +0x8a
entity.go:13: running "goen": exit status 2
```

Thanks, @lambdalisue 

I test some importers like below:

```go
package main

import (
        "fmt"
        "go/build"
        "go/importer"
        "go/token"
        "go/types"
        "golang.org/x/tools/go/packages"
        "strings"

        _ "github.com/Masterminds/squirrel"
        _ "github.com/gofrs/uuid"
        _ "github.com/lann/builder"
        _ "github.com/lann/ps"
        _ "github.com/satori/go.uuid"
)

func do(s string, imp types.Importer) {
        fmt.Printf("### %s\n", s)
        for _, pkgPath := range []string{
                "github.com/Masterminds/squirrel",
                "github.com/gofrs/uuid",
                "github.com/lann/builder",
                "github.com/lann/ps",
                "github.com/satori/go.uuid",
        } {
                fmt.Printf("import %q\n", pkgPath)
                if pkg, err := imp.Import(pkgPath); err != nil {
                        for _, line := range strings.Split(err.Error(), "\n") {
                                line = strings.TrimRight(line, "\r\n")
                                fmt.Printf("\tERROR>>> %s\n", line)
                        }
                } else {
                        fmt.Printf("\t%q\n", pkg.Name())
                }
        }
        fmt.Println()
}

type bImporter struct{}

func (*bImporter) Import(path string) (*types.Package, error) {
        bPkg, err := build.Default.Import(path, "", 0)
        if err != nil {
                return nil, err
        }
        return types.NewPackage(path, bPkg.Name), nil
}

type pImporter struct{}

func (*pImporter) Import(path string) (*types.Package, error) {
        pkgs, err := packages.Load(&packages.Config{
                Mode: packages.LoadSyntax,
        }, path)
        if err != nil {
                return nil, err
        } else if len(pkgs) != 1 {
                return nil, fmt.Errorf("%d packages containing in %q", len(pkgs), path)
        }
        pkg := pkgs[0]
        return types.NewPackage(pkg.PkgPath, pkg.Name), nil
}

func main() {
        do("default importer", importer.Default())
        do("source importer", importer.ForCompiler(token.NewFileSet(), "source", nil))
        do("build importer", &bImporter{})
        do("gc importer", importer.ForCompiler(token.NewFileSet(), "gc", nil))
        do("packages importer", &pImporter{})
}
```

under environment like below:

```bash
$ cat go.mod
module github.com/kamichidu/testing

go 1.12

require (
        github.com/Masterminds/squirrel v1.1.0
        github.com/go-sql-driver/mysql v1.4.1 // indirect
        github.com/gofrs/uuid v3.2.0+incompatible
        github.com/kr/pretty v0.1.0 // indirect
        github.com/lann/builder v0.0.0-20180802200727-47ae307949d0
        github.com/lann/ps v0.0.0-20150810152359-62de8c46ede0
        github.com/lib/pq v1.2.0 // indirect
        github.com/mattn/go-sqlite3 v1.11.0 // indirect
        github.com/satori/go.uuid v1.2.0
        golang.org/x/tools v0.0.0-20191113055240-e33b02e76616
        google.golang.org/appengine v1.6.5 // indirect
        gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
)
```

```bash
$ ls $GOPATH
bin pkg
```

then i get output:

```bash
$ go run main.go
### default importer
import "github.com/Masterminds/squirrel"
        ERROR>>> can't find import: "github.com/Masterminds/squirrel"
import "github.com/gofrs/uuid"
        ERROR>>> can't find import: "github.com/gofrs/uuid"
import "github.com/lann/builder"
        ERROR>>> can't find import: "github.com/lann/builder"
import "github.com/lann/ps"
        ERROR>>> can't find import: "github.com/lann/ps"
import "github.com/satori/go.uuid"
        ERROR>>> can't find import: "github.com/satori/go.uuid"

### source importer
import "github.com/Masterminds/squirrel"
        ERROR>>> type-checking package "github.com/Masterminds/squirrel" failed (/home/ubuntu/.go/pkg/mod/github.com/!masterminds/squirrel@v1.1.0/case.go:7:2: could not import github.com/lann/builder (type-checking package "github.com/lann/builder" failed (/home/ubuntu/.go/pkg/mod/github.com/lann/builder@v0.0.0-20180802200727-47ae307949d0/builder.go:5:2: could not import github.com/lann/ps (cannot find package "github.com/lann/ps" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/lann/ps (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/lann/ps (from $GOPATH)))))
import "github.com/gofrs/uuid"
        "uuid"
import "github.com/lann/builder"
        ERROR>>> type-checking package "github.com/lann/builder" failed (/home/ubuntu/.go/pkg/mod/github.com/lann/builder@v0.0.0-20180802200727-47ae307949d0/builder.go:5:2: could not import github.com/lann/ps (cannot find package "github.com/lann/ps" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/lann/ps (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/lann/ps (from $GOPATH)))
import "github.com/lann/ps"
        "ps"
import "github.com/satori/go.uuid"
        "uuid"

### build importer
import "github.com/Masterminds/squirrel"
        ERROR>>> cannot find package "github.com/Masterminds/squirrel" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/Masterminds/squirrel (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/Masterminds/squirrel (from $GOPATH)
import "github.com/gofrs/uuid"
        ERROR>>> cannot find package "github.com/gofrs/uuid" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/gofrs/uuid (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/gofrs/uuid (from $GOPATH)
import "github.com/lann/builder"
        ERROR>>> cannot find package "github.com/lann/builder" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/lann/builder (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/lann/builder (from $GOPATH)
import "github.com/lann/ps"
        ERROR>>> cannot find package "github.com/lann/ps" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/lann/ps (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/lann/ps (from $GOPATH)
import "github.com/satori/go.uuid"
        ERROR>>> cannot find package "github.com/satori/go.uuid" in any of:
        ERROR>>>        /home/ubuntu/local/opt/go/1.12.9/src/github.com/satori/go.uuid (from $GOROOT)
        ERROR>>>        /home/ubuntu/.go/src/github.com/satori/go.uuid (from $GOPATH)

### gc importer
import "github.com/Masterminds/squirrel"
        ERROR>>> can't find import: "github.com/Masterminds/squirrel"
import "github.com/gofrs/uuid"
        ERROR>>> can't find import: "github.com/gofrs/uuid"
import "github.com/lann/builder"
        ERROR>>> can't find import: "github.com/lann/builder"
import "github.com/lann/ps"
        ERROR>>> can't find import: "github.com/lann/ps"
import "github.com/satori/go.uuid"
        ERROR>>> can't find import: "github.com/satori/go.uuid"

### packages importer
import "github.com/Masterminds/squirrel"
        "squirrel"
import "github.com/gofrs/uuid"
        "uuid"
import "github.com/lann/builder"
        "builder"
import "github.com/lann/ps"
        "ps"
import "github.com/satori/go.uuid"
        "uuid"
```

by output above, this issue will be fixed by importing by golang.org/x/tools/go/packages.

--- a/internal/asts/asts.go
+++ b/internal/asts/asts.go
@@ -5,8 +5,11 @@ import (
 	"go/ast"
 	"go/build"
 	"path/filepath"
+	"regexp"
 	"strings"
 )
+
+var rNonWord = regexp.MustCompile(`\W`)
 
 type VisitorFunc func(ast.Node) ast.Visitor
 
@@ -102,11 +105,8 @@ func FindPkgPath(file *ast.File, pkgName string) string {
 }
 
 func AssumePkgName(pkgPath string) string {
-	pkg, err := SrcImporter.Import(pkgPath)
-	if err != nil {
-		panic(fmt.Sprintf("goen: unable to assume package name %q: %s", pkgPath, err))
-	}
-	return pkg.Name()
+	n := rNonWord.Split(pkgPath, -1)
+	return n[len(n)-1]
 }
 
 func AssumeImport(dir string) (pkgName string, pkgPath string) {

--- a/internal/asts/asts.go
+++ b/internal/asts/asts.go
@@ -104,7 +104,7 @@ func FindPkgPath(file *ast.File, pkgName string) string {
 func AssumePkgName(pkgPath string) string {
 	bpkg, err := bImport(pkgPath, ".", 0)
 	if err != nil {
-		panic(err)
+		panic(fmt.Sprintf("goen: unable to assume package name %q: %s", pkgPath, err))
 	}
 	return bpkg.Name
 }

--- a/internal/asts/asts.go
+++ b/internal/asts/asts.go
@@ -5,11 +5,8 @@ import (
 	"go/ast"
 	"go/build"
 	"path/filepath"
-	"regexp"
 	"strings"
 )
-
-var rNonWord = regexp.MustCompile(`\W`)
 
 type VisitorFunc func(ast.Node) ast.Visitor
 
@@ -105,8 +102,11 @@ func FindPkgPath(file *ast.File, pkgName string) string {
 }
 
 func AssumePkgName(pkgPath string) string {
-	n := rNonWord.Split(pkgPath, -1)
-	return n[len(n)-1]
+	bpkg, err := bImport(pkgPath, ".", 0)
+	if err != nil {
+		panic(err)
+	}
+	return bpkg.Name
 }
 
 func AssumeImport(dir string) (pkgName string, pkgPath string) {

--- a/internal/asts/cacher.go
+++ b/internal/asts/cacher.go
@@ -2,42 +2,8 @@ package asts
 
 import (
 	"go/build"
-	"go/importer"
-	"go/types"
 	"sync"
 )
-
-var (
-	SrcImporter types.Importer = &cachedImporter{
-		// importer.For is deprecated since go1.12
-		// but goen supports older go versions, ignore this line.
-		Importer: importer.For("source", nil), // nolint: staticcheck
-		cache:    map[string]*types.Package{},
-	}
-)
-
-type cachedImporter struct {
-	types.Importer
-
-	mu sync.Mutex
-
-	cache map[string]*types.Package
-}
-
-func (i *cachedImporter) Import(path string) (*types.Package, error) {
-	i.mu.Lock()
-	defer i.mu.Unlock()
-
-	if v, ok := i.cache[path]; ok {
-		return v, nil
-	}
-	tpkg, err := i.Importer.Import(path)
-	if err != nil {
-		return nil, err
-	}
-	i.cache[path] = tpkg
-	return tpkg, nil
-}
 
 type bCacheKey struct {
 	Path   string

--- a/internal/asts/parse.go
+++ b/internal/asts/parse.go
@@ -3,14 +3,13 @@ package asts
 import (
 	"fmt"
 	"go/ast"
-	"go/build"
 	"go/parser"
 	"go/token"
 )
 
 func ParsePkgPath(pkgPath string) (*ast.Package, error) {
 	// resolve directory of pkg path
-	bpkg, err := bImport(pkgPath, ".", build.FindOnly)
+	bpkg, err := bImport(pkgPath, ".", 0)
 	if err != nil {
 		return nil, err
 	}
@@ -20,7 +19,7 @@ func ParsePkgPath(pkgPath string) (*ast.Package, error) {
 	if err != nil {
 		return nil, err
 	}
-	apkg, ok := apkgs[AssumePkgName(pkgPath)]
+	apkg, ok := apkgs[bpkg.Name]
 	if !ok {
 		return nil, fmt.Errorf("goen: package directory %q is not found", pkgPath)
 	}

--- a/internal/asts/parse.go
+++ b/internal/asts/parse.go
@@ -9,13 +9,8 @@ import (
 )
 
 func ParsePkgPath(pkgPath string) (*ast.Package, error) {
-	// resolve pkgPath as a vendor if possible
-	tpkg, err := SrcImporter.Import(pkgPath)
-	if err != nil {
-		return nil, err
-	}
 	// resolve directory of pkg path
-	bpkg, err := bImport(tpkg.Path(), ".", build.FindOnly)
+	bpkg, err := bImport(pkgPath, ".", build.FindOnly)
 	if err != nil {
 		return nil, err
 	}
@@ -25,7 +20,7 @@ func ParsePkgPath(pkgPath string) (*ast.Package, error) {
 	if err != nil {
 		return nil, err
 	}
-	apkg, ok := apkgs[tpkg.Name()]
+	apkg, ok := apkgs[AssumePkgName(pkgPath)]
 	if !ok {
 		return nil, fmt.Errorf("goen: package directory %q is not found", pkgPath)
 	}

